Warning on unsecure origin

gabrieledarrigo opened the pull request:
### Description

Actually, when `sha256` is called in a not secure context a `TypeError` bubble up because  `Crypto.subtle` is undefined.
I changed the function behavior: if `Crypto.subtle` is undefined then `sha256`  returns and log a meaningful warning message.
I also added a section to the FAQ with links related to Crypto API.

### References

- https://github.com/auth0/auth0-spa-js/issues/96
- https://github.com/auth0/auth0-spa-js/issues/82

### Testing

I added a unit test to the suite to cover both the return of `sha256`  in case `Crypto.subtle`  is undefined and the console message.
If needed I can eventually split the test in two different suites, one for the function and the other for the console warning.

- [X] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [X] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used, if not `master`

gabrieledarrigo commented:
> I love this! Thanks for the PR 🎉 Can we do this check inside the `createAuth0Client` method? I think it's easier if we warn the developer in the moment they start the SDK, instead of when using one of the methods. What do you think?

Ciao @luisrudge, yes it's a good idea!
So, before proceeding, I have few questions to point me in the right direction:
If the check inside `createAuth0Client` returns false, what should we do?
Warn the user and the return early without creating the client? This seems the best scenario for me.
For example:

```
export default async function createAuth0Client(options: Auth0ClientOptions) {
  if (typeof ) {
    console.warn('Auth0 sdk etc etc...');
    return;
  }

  const auth0 = new Auth0Client(options);
  // other function's code
}
```

Then, since we are checking inside `createAuth0Client` , do you prefer to test the same condition:

```
typeof window.crypto.subtle === 'undefined'
```

Or we can test something more suitable to the case?
I was wondering that, if we keep the same condition to test, we can move the check inside a utility function ad use it inside  `createAuth0Client`.
Let me know!

luisrudge commented:
Yeah, I think you can do a console.error and just bail out. No need to continue because nothing will work. I think we can start simple for now, let's go with the typeof check. If we need to to more check in the future, we can abstract into its own function.

gabrieledarrigo commented:
That's fine!
I'll try to change the code this evening!

gabrieledarrigo commented:
Ok, I've rebased to master and moved the check inside `createAuth0Client`, adding the required tests inside `index.js` (I covered both the branch where the client can be undefined or not, let me know if you prefer to cover only the `undefined` case).
Waiting for your opinion!

luisrudge commented:
@gabrieledarrigo I made a few changes to the PR, but it looks pretty good. Thanks for the PR 🎉

gabrieledarrigo commented:
That's totally fine, glad to help 😃 !
ps: I fixed the failing test.

luisrudge commented:
OPS! forgot to change that ahahaha. Thanks!

dev-kingdom commented:
Is there any way we run it unsecure origin?

stevehobbsdev commented:
@dev-kingdom Unfortuantely not. We rely on some crypto methods that are only available on a secure origin.
